src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -54,4 +65,42 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     final protected String parserName() {
         return MatchAllQueryParser.NAME;
     }
-}
\ No newline at end of file
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        if (this.boost == 1.0f) {
+            return Queries.newMatchAllQuery();
+        }
+
+        MatchAllDocsQuery query = new MatchAllDocsQuery();
+        query.setBoost(boost);
+        return query;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof MatchAllQueryBuilder) {
+            MatchAllQueryBuilder other = (MatchAllQueryBuilder) obj;
+            return (Float.floatToIntBits(this.boost) == Float.floatToIntBits(other.boost()));
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return Float.floatToIntBits(this.boost);
+    }
```

Comment:
Looks fancy, fine if I can add some more curly brackets to the one line ifs since you urged me to do so in another PR the other day ;-)